src/main/java/org/mitre/synthea/modules/BloodPressureValueGenerator.java

```diff
@@ -47,156 +87,167 @@ public BloodPressureValueGenerator(Person person, SysDias sysDias) {
-   *
-   * @param time a timestamp which shall be covered by the generators in the buffer
-   * @param previousValueGenerator a previous generator, for potential continuity
-   */
-  private void createNewGenerators(long time) {
-    int endIndex = 0;
-    long endTime = Long.MIN_VALUE;
-    for (int i = 0; i < RING_ENTRIES; i++) {
-      if (ringBuffer[i] != null && ringBuffer[i].getEndTime() > endTime) {
-        endIndex = i;
-        endTime = ringBuffer[i].getEndTime();
-      }
+    if (personalDrugImpacts.contains(drug, sysDias)) {
+      return personalDrugImpacts.get(drug, sysDias);
     }
 
-    TrendingValueGenerator previousValueGenerator;
-    if (time - endTime < TIMETRAVEL_DURATION) {
-      // If the last ringbuffer entry is maximum of TIMETRAVEL_DURATION days in the past,
-      // then continue from it.
-      previousValueGenerator = ringBuffer[endIndex];
-    } else {
-      // Last entry is too far in the past. Start over.
-      previousValueGenerator = null;
-    }
+    // note these are intentionally flipped. "max impact" == lower number, since these are negative
+    double impact = person.rand(impactRange.high, impactRange.low);
 
-    long currentTime;
-    long generatePeriod;
-    double startValue;
-    if (previousValueGenerator == null) {
-      // There is no recent previous buffer entry. Start from a few days in the past.
-      // System.out.println("Starting over");
-      currentTime = time - TIMETRAVEL_DURATION;
-      generatePeriod = TIMETRAVEL_DURATION + TIMETRAVEL_DURATION;
-      startValue = calculateMean(person, currentTime);
-    } else {
-      // System.out.println("Continuing @ " + endTime);
-      currentTime = endTime;
-      generatePeriod = TIMETRAVEL_DURATION;
-      startValue = previousValueGenerator.getValue(endTime);
-    }
+    personalDrugImpacts.put(drug, sysDias, impact);
 
-    while (generatePeriod > 0L) {
-      final int days = minTrendDuration()
-          + person.randInt(maxTrendDuration() - minTrendDuration() + 1);
-      long duration = ONE_DAY * days;
-      double endValue;
-      do { // Limit the maximum rate of change.
-        endValue = calculateMean(person, currentTime + duration);
-      } while (Math.abs(startValue - endValue) > maxChangePerDay() * duration);
-
-      ringBuffer[ringIndex] = new TrendingValueGenerator(person, 1.0, startValue, endValue,
-          currentTime, currentTime + duration, null, null);
-      // System.out.println("Filled [" + ringIndex + "] with: " + ringBuffer[ringIndex]);
-      ringIndex++;
-      ringIndex = ringIndex % RING_ENTRIES;
-      currentTime = currentTime + duration + 1L;
-      generatePeriod -= duration;
-      startValue = endValue;
-    }
+    return impact;
   }
 
-  private double calculateMean(Person person, long time) {
-    // TODO: Take additional factors into consideration: age + gender
+  private double calculateMean(Person person, Double startValue, int days, long time) {
```

Comment:
I'm confused by `calculateMean(...)` ... it does not calculate a mean, it just seems to set a static baseline (if one has not been set already), and return that baseline.

There is no mean or even random noise added.

Also, the arguments `startValue`, `days`, and `time` are not used at all.